build_tools/azure/upload_codecov.sh

```diff
@@ -13,4 +13,6 @@ coverage combine --append
 popd
 cp $TEST_DIR/.coverage $BUILD_REPOSITORY_LOCALPATH
 
-codecov --root $BUILD_REPOSITORY_LOCALPATH -t $CODECOV_TOKEN || echo "codecov upload failed"
+curl -Os https://uploader.codecov.io/latest/linux/codecov
+chmod +x codecov
+./codecov -t ${CODECOV_TOKEN} --rootDir $BUILD_REPOSITORY_LOCALPATH || echo "codecov upload failed"
```

Comment:
I think we can remove `|| echo "codecov upload failed"`. It seems like `codecov` already does that unless we use `-Z` (from `./codecov -h`):
```
  -Z, --nonZero                 Should errors exit with a non-zero (default:
                                false)                [boolean] [default: false]
```